membership-attribute-service/test/repositories/DynamoAttributeServiceTest.scala

```diff
@@ -87,4 +87,57 @@ class DynamoAttributeServiceTest extends Specification {
       )
     }
   }
+
+  "update" should {
+    "add the attribute if it's not already in the table" in {
+      val newAttributes = Attributes(UserId = "6789", RecurringContributionPaymentPlan = Some("Monthly Contribution"))
+
+      val result = for {
+        _ <- repo.update(newAttributes)
+        retrieved <- repo.get("6789")
+      } yield retrieved
+
+      result must be_==(Some(newAttributes)).await
+    }
+
+    "update a user who has bought a digital subscription" in {
```

Comment:
@paulbrown1982 an update should update the DigitalSubscriptionExpiryDate right?

Reply:
Yup it's fine. When reading back / upon construction it takes a max of the stored and generated date when serialising it back out for Staff subs.